src/groupeddataframe/splitapplycombine.jl

```diff
@@ -19,34 +19,31 @@ function gen_groups(idx::Vector{Int})
 end
 
 function _combine_prepare(gd::GroupedDataFrame,
-                          @nospecialize(cs::Union{Pair, Base.Callable,
+                          @nospecialize(combine_operations::Union{Pair, Base.Callable,
                                         ColumnIndex, MultiColumnIndex}...);
                           keepkeys::Bool, ungroup::Bool, copycols::Bool,
                           keeprows::Bool, renamecols::Bool)
     if !ungroup && !keepkeys
         throw(ArgumentError("keepkeys=false when ungroup=false is not allowed"))
     end
 
-    cs_vec = []
-    for p in cs
+    processed_combine_ops = map(combine_operations) do p
+        if p isa Pair && first(p) isa Tuple
+            # an explicit error is thrown as this was allowed in the past
+            throw(ArgumentError("passing a Tuple $(first(p)) as column selector is not " *
+                                "supported, use a vector $(collect(first(p))) instead"))
+        end
+
         if p === nrow
-            push!(cs_vec, nrow => :nrow)
-        elseif p isa AbstractVecOrMat{<:Pair}
-            append!(cs_vec, p)
+            nrow => :nrow
         else
-            push!(cs_vec, p)
+            p
```

Comment:
this line is incorrect as you mix vector and scalar values in it. Note the distinction between `append!` and `push!` in the original code.